### PR TITLE
starkit: allow Load behavior to be modified at runtime

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

File: internal/tiltfile/starkit/fixture.go [hunk 1]
```diff
@@ -12,25 +12,37 @@ import (
...
-	fs         map[string]string
-	out        *bytes.Buffer
-	useRealFS  bool // Use a real filesystem
+	tb               testing.TB
+	extensions       []Extension
+	path             string
+	temp             *tempdir.TempDirFixture
+	fs               map[string]string
+	out              *bytes.Buffer
+	useRealFS        bool // Use a real filesystem
```
Comment (nicks) [inline internal/tiltfile/starkit/fixture.go]: this is no longer used?

File: internal/tiltfile/starkit/fixture.go [hunk 2]
```diff
@@ -12,25 +12,37 @@ import (
...
+	loadInterceptors []LoadInterceptor
+	oldDir           string
 }
 
 func NewFixture(tb testing.TB, extensions ...Extension) *Fixture {
+	temp := tempdir.NewTempDirFixture(tb)
+	oldDir, err := os.Getwd()
```
Comment (nicks) [inline internal/tiltfile/starkit/fixture.go]: can we make changing the working directory an option on TempDirFixture?

File: internal/tiltfile/starkit/environment.go [hunk 3]
```diff
@@ -146,31 +152,60 @@ func (e *Environment) start(path string) (Model, error) {
...
+		if newPath != "" {
+			// we found an interceptor that does something with this path, return early
+			return newPath, nil
+		}
+	}
+
+	return AbsPath(t, path), nil
+}
+
+func (e *Environment) doLoad(t *starlark.Thread, path string, localPath string) (starlark.StringDict, error) {
```
Comment (nicks) [inline internal/tiltfile/starkit/environment.go]: can you elaborate on why this needs two paths? afaict it only uses one

File: internal/tiltfile/starkit/environment.go [hunk 4]
```diff
@@ -146,31 +152,60 @@ func (e *Environment) start(path string) (Model, error) {
...
-	// If the path isn't absolute, the loadCache won't work correctly.
-	if !filepath.IsAbs(path) {
-		return starlark.StringDict{}, fmt.Errorf("internal error: path must be absolute")
+	localPath, err := e.getPath(t, path)
+	if err != nil {
+		e.loadCache[localPath] = loadCacheEntry{
+			status:  loadStatusDone,
+			exports: starlark.StringDict{},
+			err:     err,
+		}
```
Comment (nicks) [inline internal/tiltfile/starkit/environment.go]: i think you need to return here? can you add a test for the case where the interceptor returns an error?

File: internal/tiltfile/starkit/fixture.go (hunk 1 above)
Comment (jazzdan) [reply internal/tiltfile/starkit/fixture.go]: It is still used.

Screen Shot 2020-02-05 at 5 00 03 PM

File: internal/tiltfile/starkit/fixture.go (hunk 2 above)
Comment (jazzdan) [reply internal/tiltfile/starkit/fixture.go]: Yeah! And I would also like to socialize that out to the other tests where we do similar things.

File: internal/tiltfile/starkit/fixture.go [hunk 5]
```diff
@@ -12,23 +12,32 @@ import (
...
+	out              *bytes.Buffer
+	useRealFS        bool // Use a real filesystem
+	loadInterceptors []LoadInterceptor
 }
 
 func NewFixture(tb testing.TB, extensions ...Extension) *Fixture {
+	temp := tempdir.NewTempDirFixture(tb)
+	temp.Chdir()
+
+	_ = os.Chdir(temp.Path())
```
Comment (nicks) [inline internal/tiltfile/starkit/fixture.go]: this is obsolete now? it's covered by the temp.Chdir

Comment (jazzdan) [reply internal/tiltfile/starkit/fixture.go]: Yup thanks

**Pull request merged**